consensus/src/txn_manager.rs

```diff
@@ -69,7 +70,11 @@ impl TxnManager for MempoolProxy {
             Some(txns) => txns,
             None => return Ok(()),
         };
-        for (txn, status) in txns.iter().zip(compute_results.compute_status().iter()) {
+        // skip the block metadata txn result
+        for (txn, status) in txns
+            .iter()
+            .zip_eq(compute_results.compute_status().iter().skip(1))
```

Comment:
zip_eq to be safe?
is there another way to skip the metadata txn via looking at the txn or the outputs? arbitrarily skipping seems ... easy to introduce bugs later...

Reply:
yeah, not right now, unfortunately the compute result doesn't have any info I can link back to the original txn